EAT and JSON support (#25)
ietf-rats-wg/draft-ietf-rats-uccs: draft-ietf-rats-uccs.md

<!--  LocalWords:  Verifier's CWTs Attester Verifier FCFS
 -->

# JSON Support

The above definitions, concepts and security considerations all may be applied to define a JSON-encoded unsigned token.

> This may be referred to as a "UJSC", an Unprotected JWT Claims Set".

Reviewer: ```suggestion
This can be referred to as a "UJCS", an Unprotected JWT Claims Set".
```

> This can be referred to as a "UJCS", an Unprotected JWT Claims Set".

Reviewer 2: ```suggestion
Such an unsigned token can be referred to as a "UJCS", an "Unprotected JWT Claims Set".
```

> The CDDL in the appendix above works for JSON as well as CBOR through the use of the JC<> CDDL generic.

Reviewer 2: ```suggestion
The CDDL definition in the appendix above works for JSON as well as CBOR through the use of the JC<> CDDL generic.
```

> The CDDL definition for a UJCS token is identical with a Claims-Set.

Reviewer: ```suggestion
The definition for a UJCS token in CDDL is identical to that of a Claims-Set.
```

> The definition for a UJCS token in CDDL is identical to that of a Claims-Set.

Reviewer 2: I am having a some difficulties parsing that sentence.

Reviewer: I'm not sure what the issue is with parsing here.

_"A UJCS token is a Claims-Set with JSON encoding"_ would sound better?

Reviewer 2: Neither UCCS nore "UJCS" are token?

Maybe: The CDDL definition in Figure 1 can be used for "UJCS".

Reviewer: > Neither UCCS nore "UJCS" are token?

ok, now I can parse your parsing error :-)

> Maybe: The CDDL definition in Figure 1 can be used for "UJCS".

yeah, why not.

Reviewer 2: ```suggestion
The CDDL definition in Figure FIXME can be used for "UJCS".
```

This CDDL definition (fragment) has no Figure label, so I used "FIXME" in the suggestion above.
~~~ cddl
UJCS-Token = Claims-Set
~~~

~~~ cddl
UJCS-Token = Claims-Set
~~~

# EAT

> The following CDDL connects CBOR and JSON tokens into EAT via CDDL sockets in EAT for this purpose.

Reviewer: This doesn't read super smoothly :-)

What about:

```suggestion
The following CDDL adds UCCS and UJCS tokens to EAT using the predefined extension points.
```

> The following CDDL adds UCCS and UJCS tokens to EAT using the predefined extension points.

Reviewer 2: ```suggestion
The following CDDL definition adds types for UCCS and UJCS to EAT using its predefined extension points.
```

Reviewer 3: Indeed, UCCS and UJCS are not tokens.
Can we say "its predefined"  (to make clear that it is EAT that predefines these)?

Reviewer 2: incorporated in the suggestion above

<!--  LocalWords:  Verifier's CWTs Attester Verifier FCFS
 -->

# JSON Support

The above definitions, concepts and security considerations all may be applied to define a JSON-encoded Claims-Set.
Such an unsigned Claims-Set can be referred to as a "UJCS", an "Unprotected JWT Claims Set".
The CDDL definition in {{fig-claims-set}} can be used for a "UJCS".

~~~ cddl
UJCS = Claims-Set
~~~

> # EAT

Reviewer 3: This probably needs references, expansion, etc.

Author: Add reference to EAT submods section. That's what you are after, right?

Reviewer: The reference looks good to me ([this para](https://datatracker.ietf.org/doc/html/draft-ietf-rats-eat-22#section-4.2.18-15) of §4.2.18, specifically).

I don't think EAT needs to be expanded here, it's already mentioned in §4.1, §4.2 and §4.3.  More likely, when first introduced.

Reviewer 2: LGTM

@cabo confirmed
